src/client/pmix_client_get.c

```diff
@@ -473,6 +485,54 @@ static void infocb(pmix_status_t status,
     }
 }
 
+static pmix_status_t _getfn_fastpath(const pmix_proc_t *proc, const pmix_key_t key,
+                                     const pmix_info_t info[], size_t ninfo,
+                                     pmix_value_t **val)
+{
+    pmix_cb_t *cb = PMIX_NEW(pmix_cb_t);
+    pmix_status_t rc = PMIX_SUCCESS;
+    size_t n;
+
+    /* scan the incoming directives */
+    if (NULL != info) {
+        for (n=0; n < ninfo; n++) {
+            if (0 == strncmp(info[n].key, PMIX_DATA_SCOPE, PMIX_MAX_KEYLEN)) {
+                cb->scope = info[n].value.data.scope;
+            }
```

Comment:
Could you please put a "break" statement here? There is no point in continuing to cycle thru the directives once you have found the scope.